clients/drcachesim/tools/basic_counts.h

```diff
@@ -146,6 +175,13 @@ class basic_counts_t : public analysis_tool_t {
         std::string error;
         intptr_t last_window = -1;
         intptr_t filetype_ = -1;
+
+        // Record list of interval ids observed.
+        std::vector<uint64_t> interval_ids;
+        // Record deltas of counters seen in each trace interval.
+        std::vector<counters_t> per_interval_delta;
+        // Record cumulative counter values seen in each trace interval.
+        std::vector<counters_t> per_interval_cumulative;
```

Comment:
Seems clearer and less error-prone to have one struct with these 3 fields instead of 3 separate vectors which are assumed to align.

Reply:
This is changed in the new version now.